hilti/toolchain/include/base/optional-ref.h

```diff
@@ -0,0 +1,83 @@
+// From https://github.com/Chlorie/clu/tree/4d8bc5cfa780b9ba1d19d2e911c5bb80246a6fa4. Extended.
```

Comment:
Let's update `3rdparty/LICENSE.3rdparty`.

It would be nice if we could point out our modification, but that is probably easier said than done. If the modifications we might even be able to pull this in as is in `3rdparty/` as a full submodule (and then only use that single header). All of that should make updating this easier in the medium and long term.

Reply:
Ah, I meant to make a clean up pass over `optional-ref.h` to clean it up and see if there's  reasonable way to do the 3rd party thing. Forgot to do that, will look into it.

Reply:
As cleaned it up further, the move moved away more from the original version. Also, it feels like overkill to pull in the whole repository as a submodule just for this one pretty straight-forward file. So I went the opposite route and rewrote the header file as I'd normally written it for this code base. I think it's safe to declare this our own at this point (I still left a note for the source of inspiration in).